### PR TITLE
gasnet: build with fPIC for shared lib consumers

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gasnet/package.py [hunk 1]
```diff
@@ -60,5 +60,8 @@ def configure_args(self):
             # See the Legion webpage for details on when to/not to use.
             "--disable-pshm",
             "--with-segment-mmap-max=64MB",
+            "CC=cc -fPIC",
```
Comment (davydden) [inline var/spack/repos/builtin/packages/gasnet/package.py]: please ask compiler class for this flag.

Comment (davydden) [inline var/spack/repos/builtin/packages/gasnet/package.py]: self.compiler.pic_flag

File: var/spack/repos/builtin/packages/gasnet/package.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ class Gasnet(AutotoolsPackage):
     version('1.24.0', 'c8afdf48381e8b5a7340bdb32ca0f41a')
 
     variant('ibv', default=False, description="Support InfiniBand")
-    variant('mpi', default=False, description="Support MPI")
+    variant('mpi', default=True, description="Support MPI")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/gasnet/package.py]: @adamjstewart: That change is for you ;-)

**Comment by junghans** [comment]
Some background [here](https://bitbucket.org/berkeleylab/gasnet/pull-requests/36/wip-build-shared-libraries/diff#comment-23543607).

**Comment by junghans** [comment]
@davydden: Please review again.

File: var/spack/repos/builtin/packages/gasnet/package.py [hunk 3]
```diff
@@ -52,13 +52,18 @@ def configure_args(self):
...
+            if '+mpi' in self.spec else '--disable-mpi-compat',
             "--enable-smp-safe",
             "--enable-segment-fast",
             "--disable-aligned-segments",
             # TODO: make option so Legion can request builds with/without this.
             # See the Legion webpage for details on when to/not to use.
             "--disable-pshm",
             "--with-segment-mmap-max=64MB",
+            # for consumers with shared libs
+            "CC=cc %s" % self.compiler.pic_flag,
```
Comment (davydden) [inline var/spack/repos/builtin/packages/gasnet/package.py]: you should use `spack_cc`

File: var/spack/repos/builtin/packages/gasnet/package.py [hunk 4]
```diff
@@ -52,13 +52,18 @@ def configure_args(self):
...
+            "CXX=c++ %s" % self.compiler.pic_flag,
```
Comment (davydden) [inline var/spack/repos/builtin/packages/gasnet/package.py]: don't recall, probably `spack_cxx`

File: var/spack/repos/builtin/packages/gasnet/package.py [hunk 5]
```diff
@@ -52,13 +52,18 @@ def configure_args(self):
...
+            "MPI_CC=mpicc %s" % self.compiler.pic_flag
```
Comment (davydden) [inline var/spack/repos/builtin/packages/gasnet/package.py]: `spec['mpi'].mpicc`

**Comment by junghans** [comment]
@davydden:  [Long story](https://bitbucket.org/berkeleylab/gasnet/pull-requests/34), but in short Gasnet doesn't honoe `CFLAGS` and friends, so add it to compiler is _recommended_ way to work around that.

**Comment by davydden** [comment]
@junghans i see, thanks for clarifying this.

**Pull request merged**